src/Passes/Source/Commandline/ParameterParser.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ParameterParser.hpp"
```

Comment:
Feel free to use the same command line package we are already using here: https://github.com/microsoft/qsharp-runtime/tree/main/src/Qir/Tools/Externals/CLI11. Command line parsing is one thing where taking a dependency makes sense, since the work for nice handling including help functions etc accummulates.

Reply:
Yes, I agree. And in the long run, we will add a few dependencies, but I think this will be part of a separate PR that takes the tool closer to being fully configurable. This will also require a YAML parser so we should probably agree on what dependencies we should include overall there. I am happy to use CLI11 for parameter parsing and have a suggestion for a YAML library.